jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,43 +149,173 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 
+	private String imagesDir;
+	private boolean leaveRunning;
+	private boolean shellJob;
+	private boolean extUnixSupport;
+	private int logLevel;
+	private String logFile;
+	private boolean fileLocks;
+	private String workDir;
+
+	/**
+	 * Sets the directory that will hold the images upon dump. Corresponds to criu_set_images_dir_fd.
+	 * This must be set before calling {@link #checkPointJVM()}.
+	 * 
+	 * @param imagesDir the directory as a java.nio.file.Path
+	 * @return this
+	 * @throws NullPointerException if imagesDir is null.
+	 * @throws SecurityException if no permission to access imagesDir.
+	 * @throws IllegalArgumentException if imagesDir is not a valid directory.
+	 */
+	public CRIUSupport setImagesDir(Path imagesDir) {
+		Objects.requireNonNull(imagesDir, "Path cannot be null");
+		if (!Files.isDirectory(imagesDir)) {
+			throw new IllegalArgumentException("imagesDir is not a valid directory");
+		}
+		String dir = imagesDir.toAbsolutePath().toString();
+
+		SecurityManager manager = System.getSecurityManager();
+		if (manager != null) {
+			manager.checkWrite(dir);
+		}
+		
+		this.imagesDir = dir;
+		return this;
+	}
+
+	/**
+	 * Controls whether processes are left running after dump. Corresponds to criu_set_leave_running.
+	 * Default: false
+	 * 
+	 * @param leaveRunning
+	 * @return this
+	 */
+	public CRIUSupport setLeaveRunning(boolean leaveRunning) {
+		this.leaveRunning = leaveRunning;
+		return this;
+	}
+
+	/**
+	 * Controls ability to dump shell jobs. Corresponds to criu_set_shell_job.
+	 * Default: false
+	 * 
+	 * @param shellJob
+	 * @return this
+	 */
+	public CRIUSupport setShellJob(boolean shellJob) {
+		this.shellJob = shellJob;
+		return this;
+	}
+
+	/**
+	 * Controls whether to dump only one end of a unix socket pair. Corresponds to criu_set_ext_unix_sk.
+	 * Default: false
+	 * 
+	 * @param extUnixSupport
+	 * @return this
+	 */
+	public CRIUSupport setExtUnixSupport(boolean extUnixSupport) {
+		this.extUnixSupport = extUnixSupport;
+		return this;
+	}
+
+	/**
+	 * Sets the verbosity of log output. Corresponds to criu_set_log_level.
+	 * Default: 2 (errors + warnings)
```

Comment:
Please explain each of the levels. Do I get more or less information from larger values?